### ISSUE TITLE
CC0006 could break code when changing to foreach

### ISSUE THREAD

**Issue by wmjordan** [code-cracker/code-cracker#814]
This code:

````csharp
var a = new[] { 1 };
for (var i = 0; i < a.Length; i++)
{
    var item = a[i];
    item = 0;
}
````

Current output after fix applied (if it is a code fix bug):

````csharp
var a = new[] { 1 };
foreach (var item in a)
{
    item = 0;
}
````

We can't change the item in a for-each loop.

The fix should go to the `v1.0.x` branch.

**Comment by giggio** [comment]
Bug confirmed.

**Comment by giggio** [comment]
Closed on #831
